airflow-core/src/airflow/dag_processing/collection.py

```diff
@@ -197,6 +197,7 @@ def _serialize_dag_capturing_errors(
             session=session,
         )
         if dag_was_updated:
+            if "FabAuthManager" in conf.get("core", "auth_manager"):
             _sync_dag_perms(dag, session=session)
```

Comment:
`_sync_dag_perms` only works if FAB auth manager is configured. I think we should move the check `if "FabAuthManager" in conf.get("core", "auth_manager"):` into `_sync_dag_perms`

Reply:
And one nit. I would actually use `if isinstance(get_auth_manager(), FabAuthManager)` but this is more a personal opinion

Reply:
_sync_dag_perms was meant to be in another level under that new if...

but, that does make me wonder if we need to abstract this into the auth manager interface. It's a bit of a code smell that we have a FAB specific callout here imo.

Reply:
Yeah I wondered as well. But it is also adding something in the interface that is very specific to fab ... So I am not sure

Reply:
> I would actually use if isinstance(get_auth_manager(), FabAuthManager)

Using `isinstance` would have us handle imports for FabAuthManager, which could result in import error at times. Having it in the interface looks like a better idea